docs/PIPELINE_DEPLOYMENTS.md

```diff
@@ -0,0 +1,169 @@
+
+#### Firewall rules
+
+- [All the egress should be denied](https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#configure-firewall).
+- [Allow only Restricted API Egress by TPC at 443 port](https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#configure-firewall).
+- [Allow only Private API Egress by TPC at 443 port](https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#configure-firewall).
+- [Allow ingress Dataflow workers by TPC at ports 12345 and 12346](https://cloud.google.com/dataflow/docs/guides/routes-firewall#example_firewall_ingress_rule).
+- [Allow egress Dataflow workers by TPC at ports 12345 and 12346](https://cloud.google.com/dataflow/docs/guides/routes-firewall#example_firewall_egress_rule).
+
+#### DNS configurations
+
+- [Restricted Google APIs](https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#configure-routes).
+- [Private Google APIs](https://cloud.google.com/vpc/docs/configure-private-google-access).
+- [Restricted gcr.io](https://cloud.google.com/vpc-service-controls/docs/set-up-gke#configure-dns).
+- [Restricted Artifact Registry](https://cloud.google.com/vpc-service-controls/docs/set-up-gke#configure-dns).
+
+## Best Practice for Dataflow Flex Template Usage
+
+The *Secured Data Warehouse Module* provides resources to deploy secured Dataflow Pipeline.
+We highly recommend you to use them to deploy your Dataflow Pipeline.
+
+### Temporary and Staging Location
+
+Use the appropriated [output](../README.md#outputs) of the main module as the Temporary and Staging Location bucket in the
+[pipeline options](https://cloud.google.com/dataflow/docs/guides/setting-pipeline-options#setting_required_options):
+
+- Data ingestion project: `data_ingestion_dataflow_bucket_name`.
+- Confidential Data project: `confidential_data_dataflow_bucket_name`.
+
+### Dataflow Worker Service Account
+
+Use the appropriated [output](../README.md#outputs) of the main module as the
+[Dataflow Controller Service Account](https://cloud.google.com/dataflow/docs/concepts/security-and-permissions#specifying_a_user-managed_worker_service_account):
+
+- Data ingestion project: `dataflow_controller_service_account_email`.
+- Confidential Data project: `confidential_dataflow_controller_service_account_email`.
+
+### Customer Managed Encryption Key
+
+Use the appropriated [output](../README.md#outputs) of the main module as the [Dataflow KMS Key](https://cloud.google.com/dataflow/docs/guides/customer-managed-encryption-keys):
+
+- Data ingestion project: `cmek_data_ingestion_crypto_key`
+- Confidential project: `cmek_reidentification_crypto_key`
+
+### Disable Public IPs
+
+[Disabling Public IPs helps to better secure you data processing infrastructure.](https://cloud.google.com/dataflow/docs/guides/routes-firewall#turn_off_external_ip_address).
```

Comment:
remove the period from the hyperlink otherwise end up with ".." instead of just a single "."